src/frontend/use-replicache.ts

```diff
@@ -39,18 +46,16 @@ export function useReplicache<M extends MutatorDefs>(
     // - https://doc.replicache.dev/how-it-works#poke-optional
     // - https://github.com/supabase/realtime
     // - https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events
-    const pokeReceiver = getPokeReceiver();
-    const cancelReceiver = pokeReceiver(spaceID, async () => {
-      await r.pull();
-    });
-
+    const cancelReceiver = getPokeReceiver()(name, async () => r.pull());
     setRep(r);
 
-    return () => {
+    cleanup.current = () => {
```

Comment:
I don't understand why do this rather than return the cleanup function?

Reply:
I may be totally missing something, but without the separate cleanup `useEffect`:

1. We render the first time and call setRep.
2. State has changed, so we render again—`options` is not referentially equal to the previous `options`, so our `useEffect` runs again since `options` is a dependency.
3. Our cleanup function runs—unsubscribing/closing things.
4. `rep` is still truthy, so we bail out of the effect.
5. Everything is left closed/unsubscribed.

I've [created a sandbox](https://codesandbox.io/s/eloquent-albattani-0jc3kh?file=/src/App.js) that demonstrates it. Note that the first rerender is just React strict mode doing its thing.

By adding the separate `useEffect` with an empty dependencies array, we can be sure the cleanup function will only run on unmount. Happy to be proven wrong here (:

Reply:
I suppose since we aren't event updating the replicache instance when any of the options change due to the `if (rep) return` check we could just ignore the linter and do this:

https://codesandbox.io/s/hardcore-banach-c17x65?file=/src/App.js

Reply:
@aboodman please take another look. It seems `react-hooks/exhaustive-deps` isn't even installed/enabled? I'm not sure what I was looking at last week. As long as we don't pass options as a dependency everything works as expected.